.github/sync/sync.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
+# Usage: sync.sh <path/to/filtered>
+set -eu
+
+DOWNSTREAM="git@github.com:mrobinson/webrender.git"
+ROOT=$(pwd)
+CACHE_DIR=$ROOT/_cache
+
+# Configure git because we will be making commits.
+git config --global user.email "noreply@github.com"
+git config --global user.name "Webrender Upstream Sync"
+
+step() {
+    if [ "${TERM-}" != '' ]; then
+        tput setaf 12
+    fi
+    >&2 printf '* %s\n' "$*"
+    if [ "${TERM-}" != '' ]; then
+        tput sgr0
+    fi
+}
+
+step "Creating work directory for sync"
+mkdir -p "$1"
+cd -- "$1"
+filtered=$(pwd)
+
+step "Creating cache diretory if needed"
+mkdir -p "$CACHE_DIR"
+cd "$CACHE_DIR"
+export PATH="$PWD:$PATH"
+
+step "Downloading git-filter-repo if needed"
+if ! git filter-repo --version 2> /dev/null; then
+    curl -O https://raw.githubusercontent.com/newren/git-filter-repo/v2.38.0/git-filter-repo
+    chmod +x git-filter-repo
+
+    git filter-repo --version
+fi
+
+step "Cloning upstream if needed"
+if ! [ -e upstream ]; then
+    git clone --bare --single-branch --progress https://github.com/mozilla/gecko-dev.git upstream
+fi
+
+step "Updating upstream"
+branch=$(git -C upstream rev-parse --abbrev-ref HEAD)
+git -C upstream fetch origin $branch:$branch
+
+step "Filtering upstream"
+# Cloning and filtering is much faster than git filter-repo --source --target.
+git clone upstream -- "$filtered"
+git -C "$filtered" filter-repo --force --paths-from-file "$ROOT/.github/sync/webrender.paths"
+
+step "Adding filtered repository as a remote"
+cd "$ROOT"
+git remote add upstream "$filtered"
+git fetch upstream
```

Comment:
rename the remote to `filtered` to avoid confusing it with the raw upstream above

Reply:
Good point. I've gone with `filtered-upstream` just to be even clearer.